### PR TITLE
implement `labels` argument on `@node` and deprecates `label` and `additionalLabels`

### PR DESCRIPTION
**Pull request by a-alle**
# Description

- Implements a new `labels` argument on the `@node` directive to replace `label` and `additionalLabels` in v4. Implementation decides that `labels` takes precedence over `label`+`additionalLabels` in case both are provided.
- Adds deprecation warnings for `label` and `additionalLabels` arguments
- Adds tests
- Updates docs with deprecation warnings
- Updates v4 migration guide 

## Complexity

Complexity: Medium  - Implementation complexity is low, but many files to go through

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Made some changes to the GraphQL Toolbox? Or made some changes to the core library perhaps?

Check out your changes at https://neo4j-graphql-toolbox-stage-2825.surge.sh

Use for instance a local dbms hosted on [Neo4j Desktop](https://neo4j.com/download/) or a cloud-hosted [Aura Free](https://neo4j.com/cloud/aura-free/) dbms.

**Comment by neo4j-team-graphql** [comment]
Made some changes to the GraphQL Toolbox? Or made some changes to the core library perhaps?

Check out your changes at https://neo4j-graphql-toolbox-stage-2825.surge.sh

Use for instance a local dbms hosted on [Neo4j Desktop](https://neo4j.com/download/) or a cloud-hosted [Aura Free](https://neo4j.com/cloud/aura-free/) dbms.

**Comment by neo4j-team-graphql** [comment]
Looks like you've updated the documentation!

Check out your changes at https://neo4j-graphql-docs-2825.surge.sh

**Comment by neo4j-team-graphql** [comment]
# Performance Report

No Performance Changes


<details>
    <summary>Show Full Table</summary>
    
| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
|  aggregations.TopLevelAggregate | 3403 | 3403 | 35 | 42 | 1134 |
|  aggregations.NestedAggregation | 13514 | 13514 | 85 | 132 | 2174 |
|  aggregations.AggregationWithWhere | 10942 | 10942 | 52 | 71 | 2174 |
|  aggregations.NestedCountFromMovieToActors | 8694 | 8694 | 54 | 67 | 2174 |
|  aggregations.NestedCountFromActorsToMovie | 8900 | 8900 | 61 | 100 | 2174 |
|  aggregations.DeeplyNestedCount | 10054408 | 10054408 | 4234 | 4761 | 2008534 |
|  batch-create.BatchCreate | 3600 | 3600 | 93 | 212 | 600 |
|  connect.createAndConnect | 12419 | 12419 | 199 | 241 | 3003 |
|  connections.Connection | 13042 | 13042 | 167 | 115 | 2174 |
|  connections.NestedConnection | 33883 | 33883 | 117 | 140 | 4516 |
|  connections.ConnectionWithSort | 2277 | 2277 | 55 | 70 | 1040 |
|  connections.ConnectionWithSortAndCypher | 2277 | 2277 | 157 | 252 | 1040 |
|  create.SimpleMutation | 6 | 6 | 30 | 31 | 1 |
|  cypher-directive.TopLevelCypherDirective | 99525 | 99525 | 718 | 737 | 10168 |
|  cypher-directive.TopLevelCypherDirectiveWithColumnName | 127994 | 127994 | 308 | 377 | 10168 |
|  delete.SimpleDelete | 18361 | 18361 | 226 | 254 | 1040 |
|  delete.NestedDeleteInUpdate | 16779 | 16779 | 229 | 193 | 2040 |
|  query.SimpleQuery | 14120 | 14120 | 173 | 119 | 2174 |
|  query.QueryWhere | 8654 | 8656 | 60 | 46 | 2159 |
|  query.SimpleQueryWithNestedWhere | 8825 | 8827 | 54 | 53 | 2159 |
|  query.Nested | 10084290 | 10084290 | 9120 | 9195 | 2008534 |
|  query.NestedWithFilter | 37172 | 37172 | 103 | 208 | 4511 |
|  query.OrFilterOnRelationships | 36579 | 37028 | 151 | 341 | 1760 |
|  query.OrFilterOnRelationshipsAndNested | 30059 | 30554 | 249 | 357 | 1760 |
|  query.QueryWithNestedIn | 12597 | 12227 | 50 | 70 | 1302 |
|  query.NestedConnectionWhere | 8794 | 8794 | 47 | 67 | 2174 |
|  query.DeeplyNestedConnectionWhere | 8856 | 9054 | 104 | 79 | 2174 |
|  query.DeeplyNestedWithRelationshipFilters | 6251 | 6251 | 111 | 159 | 1134 |
|  query.DeeplyNestedWithRelationshipSingleFilters | 15438 | 15438 | 79 | 101 | 2174 |
|  query.Fulltext | 96 | 96 | 24 | 26 | 16 |
|  query.FulltextWithNestedQuery | 571 | 571 | 38 | 42 | 84 |
|  sorting.SortMultipleTypes | 2493 | 2493 | 57 | 65 | 1040 |
|  sorting.SortMultipleTypesWithCypherWithCypher | 1459 | 1451 | 154 | 209 | 1040 |
|  sorting.SortOnNestedFields | 13042 | 13042 | 83 | 90 | 2174 |
|  sorting.SortDeeplyNestedFields | 38965 | 38965 | 148 | 158 | 4516 |
|  sorting.SortWithTopLevelCypher | 2119 | 2119 | 188 | 150 | 1040 |
|  unions.SimpleUnionQuery | 321 | 321 | 106 | 75 | 35 |
|  unions.SimpleUnionQueryWithMissingFields | 293 | 293 | 89 | 109 | 35 |
|  unions.NestedUnion | 398553 | 398553 | 279 | 416 | 33033 |
|  unions.NestedUnionWithMissingFields | 372527 | 372527 | 274 | 294 | 33033 |
|  update.NestedUpdate | 2119 | 2119 | 47 | 51 | 1040 |

</details>
        
Old Schema Generation: 26.676s
Schema Generation: 26.926s

File: packages/graphql/src/classes/NodeDirective.ts [hunk 1]
```diff
@@ -48,9 +51,16 @@ export class NodeDirective {
     }
 
     public getLabels(typeName: string, context: Context): string[] {
-        const mainLabel = this.label || typeName;
-        const labels = [mainLabel, ...this.additionalLabels];
-        return this.mapLabelsWithContext(labels, context);
+        let finalLabels: string[] = [];
```
Comment (darrellwarde) [inline packages/graphql/src/classes/NodeDirective.ts:54]: `finalLabels` feels a bit confusing!

```suggestion
        let labels: string[] = [];
```

**Review by darrellwarde — CHANGES_REQUESTED** [review]
Looks good, I would personally just tweak a few things!

Toolbox link should be working now so you can check out your changes.

File: docs/modules/ROOT/pages/directives.adoc [hunk 2]
```diff
@@ -80,6 +80,9 @@ Reference: xref::type-definitions/autogeneration.adoc#type-definitions-autogener
 NOTE: The plural argument of the `@node` directive has been deprecated and will be removed in version 4.0.
 Please use the xref::directives.adoc#plural-directive[`@plural` directive] instead.
 
+NOTE: The label and additionalLabels arguments of the `@node` directive have been deprecated and will be removed in version 4.0.
```
Comment (darrellwarde) [inline docs/modules/ROOT/pages/directives.adoc:83]:
```suggestion
NOTE: The `label` and `additionalLabels` arguments of the `@node` directive have been deprecated and will be removed in version 4.0.
```

**Review by darrellwarde — DISMISSED** [review]
Looks great to me, just noticed some missing backticks in the docs but get it merged after that! 🙂

**Review by angrykoala — APPROVED** [review]

**Pull request merged**